rkt/api_service.go

```diff
@@ -0,0 +1,557 @@
+				// Only support appc image now. If it's a docker image, then it
+				// will be transformed to appc before storing in the disk store.
+				Type:    v1alpha.ImageType_IMAGE_TYPE_APPC,
+				Version: schema.AppContainerVersion.String(),
+			},
+			Id: app.Image.ID.String(),
+			// Only image format and image ID are returned in 'ListPods()'.
+		}
+
+		apps = append(apps, &v1alpha.App{
+			Name:  app.Name.String(),
+			Image: img,
+			// State and exit code are not returned in 'ListPods()'.
+		})
+	}
+
+	// Get network info.
+	var networks []*v1alpha.Network
+	for _, n := range p.nets {
+		networks = append(networks, &v1alpha.Network{
+			Name: n.NetName,
+			// There will be IPv6 support soon so distinguish between v4 and v6
+			Ipv4: n.IP.String(),
+		})
+	}
+
+	return &v1alpha.Pod{
+		Id:       p.uuid.String(),
+		Pid:      int32(pid),
+		State:    state,
+		Apps:     apps,
+		Manifest: data,
+		Networks: networks,
+	}, &manifest, nil
+}
+
+func (s *v1AlphaAPIServer) ListPods(ctx context.Context, request *v1alpha.ListPodsRequest) (*v1alpha.ListPodsResponse, error) {
+	var pods []*v1alpha.Pod
+	if err := walkPods(includeMostDirs, func(p *pod) {
+		// Get basic pod info.
+		pod, manifest, err := getBasicPod(p)
+		if err != nil { // Do not return partial pods.
+			return
+		}
+
+		// Filter the pod.
+		if !filterPod(pod, manifest, request.Filter) {
+			pods = append(pods, pod)
+		}
+	}); err != nil {
+		log.Printf("Failed to list pod: %v", err)
+		return nil, err
+	}
+	return &v1alpha.ListPodsResponse{Pods: pods}, nil
+}
+
+func (s *v1AlphaAPIServer) InspectPod(ctx context.Context, request *v1alpha.InspectPodRequest) (*v1alpha.InspectPodResponse, error) {
+	uuid, err := types.NewUUID(request.Id)
+	if err != nil {
+		log.Printf("Invalid pod id %q: %v", request.Id, err)
+		return nil, err
+	}
+
+	p, err := getPod(uuid)
+	if err != nil {
+		log.Printf("Failed to get pod %q: %v", request.Id, err)
+		return nil, err
+	}
+	defer p.Close()
+
+	// Get basic pod info.
+	pod, _, err := getBasicPod(p)
+	if err != nil {
+		return nil, err
+	}
+
+	statusDir, err := p.getStatusDir()
+	if err != nil {
+		log.Printf("Failed to get pod exit status directory: %v", err)
+		return nil, err
+	}
+
+	// Fill the extra pod info that is not available in ListPods().
+	for _, app := range pod.Apps {
+		// Fill app's image info.
+		aciInfo, err := s.store.GetACIInfoWithBlobKey(app.Image.Id)
+		if err != nil {
+			log.Printf("Failed to get ACI info for image ID %q: %v", request.Id, err)
+			return nil, err
+		}
+
+		app.Image, _, _ = s.aciInfoToV1AlphaAPIImage(aciInfo)
+
+		// Fill app's state info.
+		s, err := p.readIntFromFile(filepath.Join(statusDir, app.Name))
+		if err != nil {
+			if !os.IsNotExist(err) {
+				log.Printf("Failed to read status for app %q: %v", app.Name, err)
+				return nil, err
+			}
+			// If status file does not exit, that means the
+			// app has not exited yet.
+			app.State = v1alpha.AppState_APP_STATE_RUNNING
```

Comment:
What if the pod is in the stage "Prepared", "AbortedPrepare", etc.

It does not mean that the app is running...
```
// AppState defines the possible states of the app.
enum AppState {
        APP_STATE_UNDEFINED = 0;
        APP_STATE_RUNNING   = 1;
        APP_STATE_EXITED    = 2;
}
```
Should we add another state for "not yet started"?